### ISSUE TITLE
Save & Quit can un-pickup items

### ISSUE THREAD

**Issue by ianestrachan** [abyssonym/sai2_rando#8]
Found this while investigating https://github.com/abyssonym/sai2_rando/issues/7.

Randomizer Version: 3
Seed: 1565454364S3
Options: Fast Text Speed

A lot easier to reproduce though. You start with the Aqua Stone. Go through the Aqua gate and all the way around the edge of the map to where the Shovel normally is (the C-shaped sand island at the north boundary of the map). Pick up the item there: It's the Moon Stone. Now save and quit, go back to the title screen and load your save.

The Moon Stone is gone from your Equipment screen. If you return to the chest, it is closed again and the Moon Stone is inside once more. None of the other Stones act like this.

The trip is a slog so you can probably try turning on No Encounters, I doubt it'll make a difference.

**Comment by ianestrachan** [comment]
I've found another seed where this occurs.

Randomizer Version: 3
Seed: 1621036958S3
Options: Fast Text Speed, No Encounters, Quick Magic Switch, Quick Shovel

The route on this one is -> Silver Sword Chest (to get the Shovel) -> Chest at the start of Poka-Poka under the diggable blocks (gives you the Aqua Stone) -> Aqua Stone Gate -> Shovel location (to get the Moon Stone) -> Moon Stone Gate -> Sun Ring location (to get the Silver Sword).

Save & Quit. Reload the game. The silver sword is gone. (As a bonus, if you equip it before saving and quitting, the game still thinks you have it equipped, but if you go into the Weapons menu, the game hardlocks, presumably because the cursor is on something that doesn't exist).

If you go back to the Sun Ring location the chest will contain the Silver Sword.

I'm not sure what the connection might be between these two seeds because it's a different item, in different world map locations, and the area the chest is in is different for both.

EDIT: I did a little more testing across different emulators: I tried BSNES (Accuracy), ZSNES and SNES9x and it happened in all three. Interestingly enough, the "save & quit with silver sword equipped then open the weapon menu" freeze completely crashes ZSNES, and locks up SNES9x. BSNES handles it a little more gracefully; the controls are locked but the sound keeps playing.

**Comment by abyssonym** [comment]
Thanks for the report. I can't get to work on this right away but hopefully I'll have more time in two weeks or so.

**Comment by abyssonym** [comment]
Fixed in 875aa42cd48b0e607f62d950084f74293982ca85. I was using a certain method to give the player a random starting item in addition to Fist, etc. I had assumed that this code would only run on a new game start, but it was being run on game load as well. As a result, the randomizer had been writing a blank byte over whatever the item after the starting item was. I rewrote the patch to write an 8-bit value instead of a 16-bit value, which fixes the problem.
